crates/bevy_ecs/src/world/mod.rs

```diff
@@ -907,8 +907,10 @@ impl World {
     #[inline]
     pub fn get_resource<R: Resource>(&self) -> Option<&R> {
         let component_id = self.components.get_resource_id(TypeId::of::<R>())?;
+        let p = self.get_resource_by_id(component_id)?;
```

Comment:
This is functionally an equivalent call, but `get_resource_by_id` incurs a Send access check, which is redundant given the constraints on the `Resource` trait. This is a potential performance regression.

Reply:
While this is true, the old `get_resource_with_id` method did not have appropriate thread safety: it didn't specify this requirement, nor did it have a check.